server/subscribe.go

```diff
@@ -702,7 +702,9 @@ func (p *Plugin) hasPermissionToManageSubscription(instanceID types.ID, userID,
 				return errors.New("is not channel admin")
 			}
 		default:
-			return errors.New("can only subscribe in public and private channels")
```

Comment:
Looking at this makes me feel that subs were not allowed in DM/GM and our changes overrides that

Reply:
Yes, as requested in the [issue](https://github.com/mattermost/mattermost-plugin-jira/issues/564), the current PR changes allow the system admin to create subscriptions in DM/GM. Let me know if we need to get the approval from Doug before doing these changes, I will revert the changes

Reply:
@Kshitij-Katiyar Yes we should get a confirmation here, that the current implmentation does not allow subs in DM/GM, do we need to update that.

Also, it was mentioned in the issue that the error they are receiving is regarding permisssions and not "not allowed in DM/gM.....". Did we look into that?

Reply:
![Screenshot from 2024-09-17 10-05-07](https://github.com/user-attachments/assets/0838185a-ba53-4e5d-885d-2eea3b80170b)

This is the error mentioned in the issue, the error message is apt to the condition. Reverting the changes involving change with plugin functionality. Will add a confirmation from Doug once we create a PR on mm